py-polars/src/dataframe/general.rs

```diff
@@ -166,7 +167,7 @@ impl PyDataFrame {
     }
 
     /// set column names
-    pub fn set_column_names(&mut self, names: Vec<&str>) -> PyResult<()> {
+    pub fn set_column_names(&mut self, names: Vec<PyBackedStr>) -> PyResult<()> {
```

Comment:
How does that work? Does python need to convert to a `pybytes` first. And if so, isn't it easier to just accept `String` here?

Reply:
Most of these usages don't seem like performance hot path?